.gitlab/e2e/e2e.yml

```
@@ -27,6 +27,8 @@
     # Setup GCP credentials. https://www.pulumi.com/registry/packages/gcp/installation-configuration/
     # The service account is called `agent-e2e-tests`
     - GOOGLE_CREDENTIALS=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $E2E_TESTS_GCP_CREDENTIALS) || exit $?; export GOOGLE_CREDENTIALS
+    - export GOOGLE_CREDENTIALS_FILE=~/gcp-credentials.json
+    - echo $GOOGLE_CREDENTIALS > $GOOGLE_CREDENTIALS_FILE
```

<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
Should this be `GOOGLE_APPLICATION_CREDENTIALS`? From https://cloud.google.com/docs/authentication/provide-credentials-adc
```suggestion
    - export GOOGLE_APPLICATION_CREDENTIALS=~/gcp_credentials.json
    - $CI_PROJECT_DIR/tools/ci/fetch_secret.sh $E2E_TESTS_GCP_CREDENTIALS) >$GOOGLE_APPLICATION_CREDENTIALS || exit $?; 
```

---

Hmm. Actually, I do not know. This env variable was introduced here: https://github.com/DataDog/datadog-agent/pull/29198
We'd need some change on test-infra side if we want to change it to `GOOGLE_APPLICATION_CREDENTIALS`

---

Weird, from the linked [pulumi gcp configuration page](https://www.pulumi.com/registry/packages/gcp/installation-configuration/) I see they also use `GOOGLE_APPLICATION_CREDENTIALS`, I cannot find the variables we are using today. Though it is working. Checking with @FlorentClarret

---

`GOOGLE_CREDENTIALS` comes from https://www.pulumi.com/registry/packages/gcp/service-account/, I am updating the link in the comment

---

New suggestion, to avoid one export
```suggestion
    - GOOGLE_CREDENTIALS=$($CI_PROJECT_DIR/tools/ci/fetch_secret.sh $E2E_TESTS_GCP_CREDENTIALS) || exit $?; export GOOGLE_CREDENTIALS
```

From [pulumi doc](https://www.pulumi.com/registry/packages/gcp/service-account/)

> The GOOGLE_CREDENTIALS variable can accept either the contents or the filepath pointing at the location of your credentials file.